packages/components/src/components/Calculator/calculatorUI.tsx

```diff
@@ -99,14 +99,96 @@ export const CalculatorUI: FC<UIProps> = ({
                 {description && (
                   <div className="text-sm  text-gray-400">{description}</div>
                 )}
-                <div className="flex-grow mt-1 max-w-xs">
-                  <StyledInput
-                    value={code || ""}
-                    onChange={onChange(name)}
-                    placeholder={`Enter code for ${name}`}
-                    size="small"
-                  />
-                </div>
+                {row.tag === "text" && (
+                  <div className="flex-grow mt-1 max-w-xs">
+                    <StyledInput
+                      value={code || ""}
+                      onChange={(e) =>
+                        onChange(name)(() => {
```

Comment:
My guess is that this syntax/complexity can be simplified, but it would take more time.